daffodil-test/src/test/scala/org/apache/daffodil/section15/choice_groups/TestChoice.scala

```diff
@@ -161,7 +161,9 @@ class TestChoice {
   @Test def test_direct_dispatch_17(): Unit = { runnerCH.runOneTest("direct_dispatch_17") }
 
   // DAFFODIL-2562
-  // @Test def test_dispatch_group_choice(): Unit = { runnerCH.runOneTest("dispatch_group_choice") }
+  @Test def test_dispatch_group_choice(): Unit = {
```

Comment:
Do we really just have this one test? Is this functionality well covered by other tests?
Does this one test really prove/convince that this is all working properly?

Reply:
Definitely need to add more.  I had just gotten things to a point where it seemed I had a reasonable fix and all tests were passing.  Wanted to get this pushed up for review to make sure things looked good.  Going to add more tests now.

Reply:
I added a couple more tests just to verify that having the same choice outside of the group doesn't change anything.  I also verified that all the cases of Open/Closed match statement in TermRuntimeMixin1.scala are being hit, as that was the bulk of the changes for this commit.